.changeset/kind-dots-dress.md

```diff
@@ -0,0 +1,80 @@
+---
+"@guardian/cdk": major
+---
+
+## Reimplement reference to a pre-existing VPC
+
+This version reimplements how GuCDK references a pre-existing VPC. The changes should improve in two areas:
+1. Provide more stability by reducing the number of scenarios where the following error is thrown during synth:
+
+   ```
+   > Found an encoded list token string in a scalar string context. Use 'Fn.select(0, list)' (not 'list[0]') to extract elements from token lists.
+   ```
+
+2. Reduce the need to add values to `cdk.context.json`
+
+For most applications, these changes will be minimal with a change to the name of CloudFormation (CFN) parameters.
+
+### CloudFormation Parameter changes
+
+Previously, the CFN parameters were prefixed with the `app` being deployed.
+For example, with an `app` called `api`, GuCDK added the following CFN parameters:
+- `apiPrivateSubnets` - referencing an SSM Parameter holding the private subnets
+- `apiPublicSubnets` - referencing an SSM Parameter holding the public subnets
```

Comment:
This behaviour was introduced in https://github.com/guardian/cdk/pull/399. I don't think it makes sense to have these parameters be `app` aware. The `VpcId` parameter is not `app` aware, that is, every app in a CFN stack uses the same VPC. We also don't have a subnet per `app`.